File: plugins/fello/proj.ios/AdsFello.m
/****************************************************************************
 Copyright (c) 2013 cocos2d-x.org
 
 http://www.cocos2d-x.org
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:
 
 The above copyright notice and this permission notice shall be included in
 all copies or substantial portions of the Software.
 
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ****************************************************************************/

#import <FelloPush/KonectNotificationsAPI.h>
#import "AdsFello.h"
#import "AdsWrapper.h"

#define OUTPUT_LOG(...)     if (self.debug) NSLog(__VA_ARGS__);

@implementation AdsFello

@synthesize debug = __debug;

#pragma mark InterfaceAds impl

- (void) configDeveloperInfo: (NSMutableDictionary*) devInfo
{
    NSString* appId = (NSString*) [devInfo objectForKey:@"FelloID"];
    [KonectNotificationsAPI initialize:self launchOptions:@{} appId:appId];
}

- (void) showAds: (NSMutableDictionary*) info position:(int) pos
{
    [KonectNotificationsAPI beginInterstitial:nil];
}

- (void) hideAds: (NSMutableDictionary*) info
{
    OUTPUT_LOG(@"Fello not support hideAds!");
}

- (void) queryPoints
{
    OUTPUT_LOG(@"Fello not support query points!");
}

- (void) spendPoints: (int) points
{
    OUTPUT_LOG(@"Fello not support spend points!");
}

- (void) setDebugMode: (NSNumber*) isDebugMode
{
    self.debug = [isDebugMode boolValue];
}

- (NSString*) getSDKVersion
{
    return @"20140526";
}

- (NSString*) getPluginVersion
{
    return @"0.0.1";
}

- (void)onCompleteAdRequest:(NSString *)scene success:(BOOL)success
{
    if (success) {
        OUTPUT_LOG(@"Fello request success");
    } else {
        OUTPUT_LOG(@"Fello request failed");
    }
}

- (void)onCompleteAdRequest:(NSString *)scene reason:(NSString *)reason
{
    OUTPUT_LOG(@"Fello close: %@, %@", scene, reason);
}

- (void)onShowAd:(NSString *)scene
{
    OUTPUT_LOG(@"Fello open: %@", scene);
}

@end

File: plugins/fello/proj.android/src/org/cocos2dx/plugin/AdsFello.java
/****************************************************************************
Copyright (c) 2012-2013 cocos2d-x.org

http://www.cocos2d-x.org

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
****************************************************************************/
package org.cocos2dx.plugin;

import java.util.Hashtable;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import org.json.JSONObject;

import com.unicon_ltd.konect.sdk.KonectNotificationsAPI;
import com.unicon_ltd.konect.sdk.IKonectNotificationsCallback;

public class AdsFello implements InterfaceAds {

	private static final String LOG_TAG = "AdsFello";
	private static Activity mContext = null;
	private static boolean bDebug = false;

	protected static void LogD(String msg) {
		if (bDebug) {
			Log.d(LOG_TAG, msg);
		}
	}

	public AdsFello(Context context) {
		mContext = (Activity) context;
	}

	@Override
	public void setDebugMode(boolean debug) {
		bDebug = debug;
	}

	@Override
	public void configDeveloperInfo(Hashtable<String, String> devInfo) {
	}

	@Override
	public void showAds(Hashtable<String, String> info, int pos) {
		PluginWrapper.runOnMainThread(new Runnable() {
			@Override
			public void run() {
          KonectNotificationsAPI.beginInterstitial(null);
			}
		});
	}

	@Override
	public void hideAds(Hashtable<String, String> info) {
		LogD("Fello not support hideAds!");
	}

	@Override
	public String getSDKVersion() {
		return "20140501";
	}

	@Override
	public String getPluginVersion() {
		return "0.0.1";
	}

	@Override
	public void queryPoints() {
		LogD("Fello not support query points!");
	}

	@Override
	public void spendPoints(int points) {
		LogD("Fello not support spend points!");
	}

}
